delete-publication/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java

```diff
@@ -0,0 +1,104 @@
+package no.unit.nva.publication.delete;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.unit.nva.api.PublicationResponse;
+import no.unit.nva.model.Publication;
+import no.unit.nva.publication.exception.InputException;
+import no.unit.nva.publication.service.PublicationService;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import no.unit.nva.testutils.TestHeaders;
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.handlers.GatewayResponse;
+import nva.commons.utils.Environment;
+import nva.commons.utils.JsonUtils;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.zalando.problem.Problem;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.UUID;
+
+import static java.util.Collections.singletonMap;
+import static nva.commons.handlers.ApiGatewayHandler.ALLOWED_ORIGIN_ENV;
+import static nva.commons.utils.JsonUtils.objectMapper;
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+public class DeletePublicationHandlerTest {
```

Comment:
We need tests that go through the whole thing. I.e. a handler connected to a service with a local database and sending queries. The tests do not describe the behavior of the handler.

* Does the handler return accepted when the publication is Draft?
* What does the handler return when nhe publicaiton is Public?
* What does it return when the publication does not exist?
* What does it return when the publication is already Drafted for Deletion?
* What does it return when the user is not the owner?   

I think these tests should be in this test class. But I need Rurik's opinion on this

Reply:
Agreed